expectation.messagematch.go

```diff
@@ -0,0 +1,192 @@
+package testkit
+
+import (
+	"errors"
+
+	"github.com/dogmatiq/configkit/message"
+	"github.com/dogmatiq/dogma"
+	"github.com/dogmatiq/testkit/envelope"
+	"github.com/dogmatiq/testkit/fact"
+	"github.com/dogmatiq/testkit/internal/inflect"
+)
+
+// ToExecuteCommandMatching returns an expectation that passes if a command is
+// executed that satisfies the given predicate function.
+//
+// Always prefer using ToExecuteCommand() instead, if possible, as it provides
+// more meaningful information in the result of a failure.
+//
+// desc is a human-readable description of the expectation. It should be phrased
+// as an imperative statement, such as "debit the customer".
+//
+// pred is the predicate function. It is called for each executed command. It
+// must return nil at least one for the expectation to pass.
```

Comment:
Something not quite right with this last sentence. Should "one" be "once"?
```suggestion
// must return nil at least once for the expectation to pass.
```